### PR TITLE
add `lotus-miner storage-deals list --format=json` with transfers

### PR DESCRIPTION
**Pull request by nonsense**
This PR is updating the `lotus-miner storage-deals list` cmd to print out storage deals together with their respective transfers, for easier debugging - only when using `--format=json`. At the moment fetching storage deals and transfers is done via two separate commands and somewhat cumbersome to use given the tab-delimited tables.

`--format=json` prints out results in line-delimited JSON, which I think is much easier to read than the `table` approach we use in many commands, because:

1. Field type is right next to every value (given that we generally have 100s-1000s of deals):
```
{"datetime":"2021-09-10T00:57:41+03:00","verified-deal":true,"proposal-cid":"bafyreif5yiywi6njej3nz2uwrj6yo2dnrc7mnokk3qhhfc6m46efpurmay","deal-id":0,"deal-status":"StorageDealTransferring","client":"f3vnq2cmwig3qjisnx5hobxvsd4drn4f54xfxnv4tciw6vnjdsf5xipgafreprh5riwmgtcirpcdmi3urbg36a","piece-size":"16GiB","price":"0 FIL","duration-epochs":1494720,"transfer-id":1631214107052231430,"transfer-status":"Ongoing","transferred-data":"0B"}
```

2. Searching and formatting is very easy with `grep` and `jq`:
```
$ LOTUS_MARKETS_PATH=~/markets-repo-location ./lotus-miner storage-deals --format=json | grep 7vsgutuiem | jq
{
  "datetime": "2021-09-10T11:36:48+03:00",
  "verified-deal": true,
  "proposal-cid": "bafyreie2e66s3ztkdrfoqxm7hrch5d75i3nbrtp6jnsdcyrr7vsgutuiem",
  "deal-id": 0,
  "deal-status": "StorageDealTransferring",
  "client": "f3vnq2cmwig3qjisnx5hobxvsd4drn4f54xfxnv4tciw6vnjdsf5xipgafreprh5riwmgtcirpcdmi3urbg36a",
  "piece-size": "8GiB",
  "price": "0 FIL",
  "duration-epochs": 1494720,
  "transfer-id": 1631214107052231700,
  "transfer-status": "Ongoing",
  "transferred-data": "0B"
}
```

### CODE REVIEW

**Comment by dirkmc** [comment]
Thanks for putting this together 👍 

I agree the JSON output is clearer and easier to parse.

Would it make sense to add the output format and a flag for whether to include deal transfers onto the existing `list` command?
Something like
`./lotus-miner storage-deals list --format=json --with-transfers`

**Comment by nonsense** [comment]
> Would it make sense to add the output format and a flag for whether to include deal transfers onto the existing list command? Something like
./lotus-miner storage-deals list --format=json --with-transfers

I actually started with that, and it gets messy very quickly to support --with-transfers in the `--format=table` so I decided to just go with a separate command and keep things simpler. I don't have hard feelings about that though, so I could bundle it together with `list`, but maybe add `--with-transfers` only for `--format=json`? WDYT?

**Comment by nonsense** [comment]
It is quite hard to support `--watch` also for `data transfers`, so I opted for:
1. Not adding `--with-transfers` flag at all - always try to load transfers when using `--format=json`
2. Leave `--format=table` as is, as we might want to remove `--watch` altogether.

File: cmd/lotus-miner/market.go [hunk 1]
```diff
@@ -396,63 +403,74 @@ var dealsListCmd = &cli.Command{
...
+			return listDealsWithJSON(cctx)
 		}
-		defer closer()
 
-		ctx := lcli.DaemonContext(cctx)
+		return fmt.Errorf("unknown format: %s; use `table` or `json`", cctx.String("format"))
+	},
+}
+
+func listDealsWithTable(cctx *cli.Context) error {
```
Comment (nonsense) [inline cmd/lotus-miner/market.go:417]: For easier review - this is not modified.

**Review by dirkmc — APPROVED** [review]
LGTM 👍

**Comment by noch3** [comment]
In #7235 I used a `--json` flag. Should there be a standard approach to this? I'd like to point out that AFAIK the table output doesn't actually use TSV. The tabs get converted to spaces. This makes parsing it harder than it should be IMO.

**Comment by nonsense** [comment]
> In #7235 I used a `--json` flag. Should there be a standard approach to this? I'd like to point out that AFAIK the table output doesn't actually use TSV. The tabs get converted to spaces. This makes parsing it harder than it should be IMO.

Yes, ideally we should be consistent across various command line tools. Having said that I don't have any preference over `--format=json` or just `--json` and `--format=table` and `--table`, but probably better to be explicit.

For now I will leave the other reviewers to chime in.

**Review by magik6k — APPROVED** [review]

**Pull request merged**